### ISSUE TITLE
Some Training Problems with last version

### ISSUE THREAD

**Issue by caki35** [RangiLyu/nanodet#236]
Hello. I trained many models using previous versions of nanodet repo in which glf_head.py was used instead of one_stage_detector.py . However, I run into some problems when I started to use the last version.

1-) The train keeps starting from 1. epoch skipping warmup process. 
I take the following warning:
/home/rap019/.local/lib/python3.8/site-packages/torch/nn/functional.py:3103: UserWarning: The default behavior for interpolate/upsample with float scale_factor changed in 1.6.0 to align with other frameworks/libraries, and now uses scale_factor directly, instead of relying on the computed output size. If you wish to restore the old behavior, please set recompute_scale_factor=True. See the documentation of nn.Upsample for details.
  warnings.warn("The default behavior for interpolate/upsample with float scale_factor changed "

2-) Although I set val_intervals as 1, the repo performs validation at different epoch:

Epoch:1
...
Epoch:2
....
Epoch:8
...
Epoch:9
...
Epoch:10
...
Epoch:17
...
Epoch:20
...
Epoch:25
...
Epoch:45
...

3-) I modulated the trainer.py such that the weights are saved at each epoch. It always worked in the previous repo. Now, it saves only model_best.ckpt (not .pth) since it never enters the following for loop
"for epoch in range(start_epoch, self.cfg.schedule.total_epochs + 1):"

Lastly, is there any difference between .pth and .ckpt extensions? Because I got all models with .pth extension in the previous version of repo.

Thanks.

**Comment by RangiLyu** [comment]
1-) This warning is about bilinear interpolation behavior in PyTorch, don't worry about it.

2-) Do you mean the validation interval is not 1 when setting `val_intervals: 1`? It looks so weird. I'll check it later.

3-) nanodet/trainer/trainer.py is used for the old training script(tools/deprecated/train.py). After v0.3.0 the trainer has been replaced with PyTorch Lightning trainer. Now you need to modify tools/deprecated/task.py to change the training process. Also, .pth and .ckpt are both checkpoint files during training. Since so many people asking about why the .pth checkpoint file is larger than the model's weights, I have to change the extension to prevent misunderstanding.

**Comment by caki35** [comment]
I solve 2. and 3. issues by modifying tools/deprecated/task.py.  Actually, it performed validation but only wrote results when it achieves the current best result. 
```
if metric > self.save_flag:
            self.save_flag = metric
            best_save_path = os.path.join(self.cfg.save_dir, 'model_best')
            mkdir(self.local_rank, best_save_path)
            self.trainer.save_checkpoint(os.path.join(best_save_path, "model_{}_best.ckpt".format(self.current_epoch+1)))
            txt_path = os.path.join(best_save_path, "eval_results.txt")
            if self.local_rank < 1:
                with open(txt_path, "a") as f:
                    f.write("Epoch:{}\n".format(self.current_epoch+1))
                    for k, v in eval_results.items():
                        f.write("{}: {}\n".format(k, v))
        else:
            best_save_path = os.path.join(self.cfg.save_dir, 'model_best')
            mkdir(self.local_rank, best_save_path)
            self.trainer.save_checkpoint(os.path.join(best_save_path, "model_{}.ckpt".format(self.current_epoch+1)))
            txt_path = os.path.join(best_save_path, "eval_results.txt")
            if self.local_rank < 1:
                with open(txt_path, "a") as f:
                    f.write("Epoch:{}\n".format(self.current_epoch+1))
                    for k, v in eval_results.items():
                        f.write("{}: {}\n".format(k, v))           
            #warnings.warn('Warning! Save_key is not in eval
```

Thanks
